cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -61,7 +62,7 @@
   public HBaseMetricsTable(DatasetContext datasetContext, DatasetSpecification spec,
                            Configuration hConf, HBaseTableUtil tableUtil) throws IOException {
     this.tableUtil = tableUtil;
-    this.tableId = TableId.from(datasetContext.getNamespaceId(), spec.getName());
+    this.tableId = tableUtil.createHTableId(Id.Namespace.from(datasetContext.getNamespaceId()), spec.getName());
```

Comment:
I like this standardization. however, now can we somehow make sure that a ``TableId`` object can never be created without going through ``HBaseTableUtil``?